src/main/java/seedu/address/logic/parser/add/AddProfileCommandParser.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.logic.parser.add;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.clisyntax.ProfileCliSyntax.PREFIX_CATEGORY;
+import static seedu.address.logic.parser.clisyntax.ProfileCliSyntax.PREFIX_DESCRIPTORS;
+import static seedu.address.logic.parser.clisyntax.ProfileCliSyntax.PREFIX_TITLE;
+import static seedu.address.logic.parser.util.Util.arePrefixesPresent;
+
+import java.util.Set;
+
+import seedu.address.logic.commands.add.AddProfileCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.logic.parser.util.ProfileParserUtil;
+import seedu.address.model.profile.Descriptor;
+import seedu.address.model.profile.ProfileItem;
+import seedu.address.model.profile.ProfileItemCategory;
+import seedu.address.model.profile.Title;
+
+
+/**
+ * Parses input arguments and creates a new AddProfileCommand object
+ */
+public class AddProfileCommandParser implements Parser<AddProfileCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddProfileCommand
+     * and returns an AddProfileCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddProfileCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_TITLE, PREFIX_CATEGORY, PREFIX_DESCRIPTORS);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_TITLE, PREFIX_CATEGORY)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddProfileCommand.MESSAGE_USAGE));
+        }
```

Comment:
just a suggestion, @line 38-39
from GitHub this line looks quite long, you can consider having 
boolean b1 = ...
boolean b2 = ...
if (b1 || b2)